pkg/storage/stores/tsdb/multi_file_index_test.go

```diff
@@ -154,3 +154,60 @@ func TestMultiIndex(t *testing.T) {
 		require.Equal(t, expected, xs)
 	})
 }
+
+func BenchmarkSliceForParallelism(b *testing.B) {
+	cases := []LoadableSeries{
+		{
+			Labels: mustParseLabels(`{foo="bar"}`),
+			Chunks: []index.ChunkMeta{
+				{
+					MinTime:  0,
+					MaxTime:  3,
+					Checksum: 0,
+				},
+				{
+					MinTime:  1,
+					MaxTime:  4,
+					Checksum: 1,
+				},
+				{
+					MinTime:  2,
+					MaxTime:  5,
+					Checksum: 2,
+				},
+			},
+		},
+		{
+			Labels: mustParseLabels(`{foo="bar", bazz="buzz"}`),
+			Chunks: []index.ChunkMeta{
+				{
+					MinTime:  1,
+					MaxTime:  10,
+					Checksum: 3,
+				},
+			},
+		},
+		{
+			// should be excluded due to bounds checking
+			Labels: mustParseLabels(`{foo="bar", bazz="bozz", bonk="borb"}`),
+			Chunks: []index.ChunkMeta{
+				{
+					MinTime:  8,
+					MaxTime:  9,
+					Checksum: 4,
+				},
+			},
+		},
+	}
+
+	dir := b.TempDir()
```

Comment:
Remember to clean this up at the end

Reply:
Temporary directories created with `*testing.T` or `*testing.B` are automatically cleaned up.

```
// TempDir returns a temporary directory for the test to use.
// The directory is automatically removed by Cleanup when the test and
// all its subtests complete.
```